pkgs/by-name/dr/draupnir/package.nix

```
@@ -0,0 +1,96 @@
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, nodejs
+, matrix-sdk-crypto-nodejs
+, mkYarnPackage
+, fetchYarnDeps
+, nixosTests
+}:
+
+# docs: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#yarn2nix-javascript-yarn2nix
+let
+  hashesFile = builtins.fromJSON (builtins.readFile ./hashes.json);
+in
+  mkYarnPackage rec {
+    pname = "draupnir";
+    version = "2.0.0-beta.4";
+    src = fetchFromGitHub {
+      owner = "the-draupnir-project";
+      repo = "Draupnir";
+      rev = "v${version}";
+      hash = "sha256-ZbAst3XaTdJxCPsgPc8cAqN0lo6vOvFphqBDjduQ/to=";
+    };
+
+    nativeBuildInputs = [
+      makeWrapper
+    ];
+
+    offlineCache = fetchYarnDeps {
+      name = "${pname}-yarn-offline-cache";
+      yarnLock = src + "/yarn.lock";
+      hash = hashesFile.yarn_offline_cache_hash;
+    };
+
+    yarnLock = src + "/yarn.lock";
+    packageJSON = ./package.json;
+
+    #prebuild phase
+    preBuild = ''
+      # copy built modules to package...
+      echo "Copying built matrix-sdk-crypto-nodejs modules to package..."
+      cp -a ${matrix-sdk-crypto-nodejs}/lib/node_modules/* node_modules/
+      echo "Adding version.txt..."
+      mkdir -p deps/draupnir/
+      echo "${version}-nix" > deps/draupnir/version.txt
+    '';
+
+    buildPhase = ''
+      runHook preBuild
+      echo "Building..."
+      yarn --offline --verbose build
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share
+      cp -a . $out/share/draupnir
+
+      makeWrapper ${nodejs}/bin/node $out/bin/draupnir \
+        --add-flags $out/share/draupnir/deps/draupnir/lib/index.js
+
+      runHook postInstall
+    '';
+    distPhase = "true";
+
+    passthru = {
+      tests = {
+        inherit (nixosTests) draupnir;
+      };
+      updateScript = ./update.sh;
```

Is there anything special here, that `nix-update-script` couldn't handle? It can also deal with updating the Yarn cache hash in the derivation.

```suggestion
      updateScript = nix-update-script { };
```

---

Hashes are stored in a separate file, I think the reason was that nix-update-script couldn't handle the Yarn cache hash? I'd have to test this once a new Draupnir version comes out.